mkosi/backend.py

```diff
@@ -471,10 +471,9 @@ def flatten(lists: Iterable[Iterable[T]]) -> list[T]:
     return list(itertools.chain.from_iterable(lists))
 
 
-def current_user_uid_gid() -> tuple[int, int]:
+def current_user() -> pwd.struct_passwd:
```

Comment:
I'm not sure we should expose the raw `struct_passwd`. In most places we only use one or two things, but it litters the code with the struct attribute names, somewhat obscuring what is happening.

Reply:
I think this is fine, it's more descriptive in whats being returned

Reply:
I'm not a fan of the `pw_` prefix. However, most of the struct attributes are used (`pw_uid`, `pw_gid`, `pw_name`, `pw_dir`).

Reply:
> I think this is fine, it's more descriptive in whats being returned

It's more descriptive here, but I think the surrounding code where it calls becomes a bit harder to read.

The only use sites of `pw_dir` below would become
```python
        if path.is_relative_to("~") and os.getuid() != invoking_user_uid():
            path = Path(f"~{invoking_user_name}") / path.relative_to("~")
        path = path.expanduser()
```
for some 
```python
def invoking_user_uid() -> int:
    return int(os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID") or os.getuid())

def invoking_user_uid_gid() -> tuple[int, int]:
    uid = invoking_user_uid()
    gid = int(os.getenv("SUDO_GID") or pwd.getpwuid(uid).pw_gid)
    return uid, gid

def invoking_user_name() -> str:
    return os.getenv("SUDO_USER") or os.getenv("USER") or os.getlogin()
```

Maybe more generally:
```python
class InvokingUser:
    @staticmethod
    def uid() -> int:
         return int(os.getenv("SUDO_UID") or os.getenv("PKEXEC_UID") or os.getuid())
    @classmethod
    def gid(cls) -> int:
        return int(os.getenv("SUDO_GID") or pwd.getpwuid(cls.uid()).pw_gid)
    @staticmethod
    def name() -> str:
        return os.getenv("SUDO_USER") or os.getenv("USER") or os.getlogin()
    @classmethod
    def home(cls) -> Path:
        if p := (os.getenv("SUDO_HOME") or os.getenv("HOME")):
            return Path(p)
        return Path(f"~{cls.name()}").expanduser()
    @classmethod
    def is_running_user(cls) -> bool:
         return cls.uid == os.getuid()
```

Reply:
I like the approach with a class, but I'm less a fan of going through the other `SUDO_`* environment variables, because they do not work when pkexec is used (name would be root and home would be /root). Or to be correct, they will not work once `bin/mkosi` is dropped.